internal/controller/controllers/agent_controller.go

```diff
@@ -516,6 +506,50 @@ func (r *AgentReconciler) shouldApproveCSRsForAgent(ctx context.Context, agent *
 	return false, nil
 }
 
+func marshalNodeLabels(nodeLabels map[string]string) (string, error) {
+	b, err := json.Marshal(&nodeLabels)
+	if err != nil {
+		return "", err
+	}
+	return string(b), err
+}
+
+func (r *AgentReconciler) applyDay2NodeLabels(ctx context.Context, log logrus.FieldLogger, agent *aiv1beta1.Agent, client spoke_k8s_client.SpokeK8sClient) error {
+	if !funk.IsEmpty(agent.Spec.NodeLabels) {
+		node, err := r.getNode(agent, client)
+		if err != nil {
```

Comment:
There are couple of places we doing same thing while checking if node is ready, can we make some function for it?
i mean that we check do getNode, check it exists in at least 2 different places already, maybe we can have function for it? if possible though

Reply:
Actually seems like we can get node before this function and provide it in case it exists, same with other places

Reply:
Changed to get node from a single location